GW2EIEvtcParser/EIData/ProfHelpers/Ranger/DruidHelper.cs

```diff
@@ -19,6 +20,19 @@ internal static class DruidHelper
             new EffectCastFinder(GlyphOfEquality, EffectGUIDs.DruidGlyphOfEquality).UsingSrcSpecChecker(Spec.Druid)
         };
 
+        public static IReadOnlyList<AnimatedCastEvent> ComputeAncestralGraceCastEvents(Player druid, CombatData combatData, SkillData skillData, AgentData agentData)
+        {
+            var res = new List<AnimatedCastEvent>();
+            SkillItem skill = skillData.Get(AncestralGraceSkill);
+            var applies = combatData.GetBuffData(AncestralGraceEffect).OfType<BuffApplyEvent>().Where(x => x.To == druid.AgentItem).ToList();
+            var removals = combatData.GetBuffData(AncestralGraceEffect).OfType<BuffRemoveAllEvent>().Where(x => x.To == druid.AgentItem).ToList();
+            for (int i = 0; i < applies.Count && i < removals.Count; i++)
+            {
+                res.Add(new AnimatedCastEvent(druid.AgentItem, skill, applies[i].Time, applies[i].AppliedDuration - removals[i].RemovedDuration));
```

Comment:
Replace "applies[i].AppliedDuration - removals[i].RemovedDuration" with "removals[i].Time - applies[i].Time" to be more robust.